codeship-steps.yml

```diff
@@ -1,123 +1,142 @@
-            - name: test-internal-dependencies
-              service: testing
-              command: './packages/devops/scripts/ci/testInternalDependencies.sh'
-            - name: test-web-frontend
-              service: testing
-              command: './packages/devops/scripts/ci/testFrontend.sh web-frontend'
+        - name: Test admin-panel-server
+          command: './packages/devops/scripts/ci/testBackend.sh admin-panel-server'
+        - name: Test lesmis-server
+          command: './packages/devops/scripts/ci/testBackend.sh lesmis-server'
+        - name: Test meditrak-server
+          command: './packages/devops/scripts/ci/testBackend.sh meditrak-server'
+        - name: Test web-config-server
+          command: './packages/devops/scripts/ci/testBackend.sh web-config-server'
+        - name: Test psss-server
+          command: './packages/devops/scripts/ci/testBackend.sh psss-server'
+        - name: Test report-server
+          command: './packages/devops/scripts/ci/testBackend.sh report-server'
+        - name: Test entity-server
+          command: './packages/devops/scripts/ci/testBackend.sh entity-server'
+        - name: Test database
+          command: './packages/devops/scripts/ci/testBackend.sh database'
+        - name: Test data-api
+          command: './packages/devops/scripts/ci/testBackend.sh data-api'
+        - name: Test auth
+          command: './packages/devops/scripts/ci/testBackend.sh auth'
+        - name: Test indicators
+          command: './packages/devops/scripts/ci/testBackend.sh indicators'
+        - name: Test internal-dependencies
+          command: './packages/devops/scripts/ci/testInternalDependencies.sh'
+        - name: Test web-frontend
+          command: './packages/devops/scripts/ci/testFrontend.sh web-frontend'
+
+- name: Check for deployment
+  service: deployment
+  command: './packages/devops/scripts/ci/checkDeploymentExists.sh'
 
 - type: serial
-  name: pull latest into ec2
+  name: Set up ssh key
   service: deployment
   exclude: -e2e$
   steps:
-    - name: reinstate SSH Private Key File
+    - name: Reinstate SSH Private Key File
       # Need to use 'sed' here to remove open/close quotes, see: https://github.com/docker/compose/issues/2854
       # Note: this issue is fixed in more recent versions of docker-compose (1.26+) but codeship still uses an older version
       command: /bin/bash -c "echo -e $PRIVATE_SSH_KEY | sed -e 's/^"//' -e 's/"$//' >> /root/.ssh/id_rsa"
     - name: chmod id_rsa
       command: chmod 600 /root/.ssh/id_rsa
-    - name: pull latest and install dependencies
-      command: './packages/devops/scripts/ci/pullLatest.sh'
+
 - type: serial
-  name: deploy-all
-  service: deployment
+  name: Build front ends
+  service: builder
   exclude: -e2e$
   steps:
-    - type: serial
-      name: deploy-backends
+    # front end builds are split into batches as they consume a lot of memory
```

Comment:
Despite having four cores on the machine now, and 7.5GB memory, the front ends still seemed slightly flakey when run in parallel. When https://github.com/beyondessential/tupaia/pull/3076 is merged I will try again with all four in parallel 🤞